libraries/chain/database.cpp

```diff
@@ -3971,6 +3972,20 @@ namespace golos {
             }
         }
 
+        void database::clear_expired_delegations() {
+            auto now = head_block_time();
+            const auto& delegations_by_exp = get_index<vesting_delegation_expiration_index, by_expiration>();
+            auto itr = delegations_by_exp.begin();
+            while (itr != delegations_by_exp.end() && itr->expiration < now) {
+                modify(get_account(itr->delegator), [&](account_object& a) {
+                    a.delegated_vesting_shares -= itr->vesting_shares;
+                });
+                push_virtual_operation(return_vesting_delegation_operation(itr->delegator, itr->vesting_shares));
```

Comment:
It means right in here an expiring delegation would be increased at ```account_object```? (In case of ```itr->vesting_shares``` is a negative by default)

Reply:
`itr->vesting_shares` is always positive, `delta` (from https://github.com/GolosChain/golos/pull/585#discussion_r184392121) can be negative, and `o.vesting_shares = -delta;` makes it positive.

Reply:
My fault. It is fine.